src/sources/datadog_agent/traces.rs

```diff
@@ -178,14 +178,18 @@ fn convert_dd_tracer_payload(payload: ddtrace_proto::TracerPayload) -> Vec<Trace
             let mut trace_tags = convert_tags(trace.tags);
             trace_tags.extend(tags.clone());
             trace_event.insert("tags", Value::from(trace_tags));
-            trace_event.insert(
-                "spans",
-                trace
-                    .spans
-                    .into_iter()
-                    .map(|s| Value::from(convert_span(s)))
-                    .collect::<Vec<Value>>(),
-            );
+
+            let n_spans = trace.spans.len();
+
+            let spans = trace
+                .spans
+                .into_iter()
+                .map(|s| Value::from(convert_span(s)))
+                .collect::<Vec<Value>>();
+
+            assert_eq!(n_spans, spans.len(), "Not all spans converted!");
```

Comment:
I don't think this is common.
This was from before I found root cause.
I was suspicious that perhaps we were dropping spans (which would throw off the stats calculation) silently.
This was because at the time when I was looking at the conversion from BTreeMap -> Value , I thought I saw a comment there saying that it is possible not everything will be converted. But I'm not finding it now :thinking:  I might have been mistaken because looking at it now, I'm not seeing how the conversion from BTreeMap<String, Value> -> Value can fail.

In which case, I will just revert this change.